lib/ecto/query/builder/select.ex

```diff
@@ -348,6 +362,9 @@ defmodule Ecto.Query.Builder.Select do
   defp merge(query, select, old_expr, old_params, old_subqueries, old_take, old_aliases, new_select) do
     %{expr: new_expr, params: new_params, subqueries: new_subqueries, take: new_take, aliases: new_aliases} = new_select
 
+    {old_expr, old_take} = expand_map!(query, old_expr, old_take)
+    {new_expr, new_take} = expand_map!(query, new_expr, new_take)
```

Comment:
Instead of expand it here, I think you could say in classify_merge that it has type `{:map, meta, :runtime}`.

Reply:
Oh yeah I tried this but then it makes it less powerful than `map/2`. For instance you can do this with map/2:

`from c in Comment, select: map(c, [:id]), select_merge: [:dislikes]`

But if I don't expand `map/1` here it will complain that a source can't be merged into a map when doing this

`from c in Comment, select: map(c), select_merge: [:dislikes]`

Reply:
I could make it something similar `{:map, meta, something_about_map/1}` to catch this case?